djvm/src/test/kotlin/net/corda/djvm/ExceptionWithoutSyntheticCounterpartTest.kt

```diff
@@ -0,0 +1,35 @@
+package net.corda.djvm
+
+import net.corda.djvm.SandboxType.KOTLIN
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.ValueSource
+
+/**
+ * Check that the DJVM does not create synthetic throwable
+ * counterparts for these exceptions, because we're going
+ * to use the JVM's own [Throwable] classes here.
+ *
+ * This list is representative rather than exhaustive.
+ */
+class ExceptionWithoutSyntheticCounterpartTest : TestBase(KOTLIN) {
+    @ParameterizedTest
+    @ValueSource(strings = [
+        "sandbox.java.lang.Throwable",
+        "sandbox.java.lang.Exception",
+        "sandbox.java.lang.RuntimeException",
+        "sandbox.java.lang.Error",
+        "sandbox.java.lang.LinkageError",
+        "sandbox.java.lang.ThreadDeath",
+        "sandbox.java.lang.VirtualMachineError",
+        "sandbox/java/lang/DJVMThrowableWrapper"
```

Comment:
Shouldn't `loadClass` take the binary name of the class? Why are we passing `/` instead of `.`?
I would expect `loadClass` to never find a class when not in the binary form.

Reply:
Yikes! You are correct! _Doh!_ :1st_place_medal:

Reply:
The test is expecting `classLoader` to throw `ClassNotFoundException`, which it does, but we're not trying to load the correct class, since the binary form isn't being used.

Reply:
But _then_ it tries to load another class, which apparently _does_ still exist...

Reply:
The first time it loads the class it works, but the second time it fails:
```
loader (instance of  net/corda/djvm/rewiring/SandboxClassLoader): attempted  duplicate class definition for name: "sandbox/java/lang/DJVMThrowableWrapper"
```

Reply:
The `SandboxClassLoader.loadClass()`method seems to be "forgiving" of this kind of error. Hmm...